bin/check_hcid.py

```diff
@@ -76,37 +75,42 @@ def check_report_for_hcid(hcid_dict, taxonomy_dir, kreport_file):
             hcid_dict[taxid]["classified_parent_found"] = True
 
 
-def map_to_refs(query, reference, preset):
+def map_to_refs(query, ref_sam):
     counts = defaultdict(int)
     ranges = defaultdict(list)
     read_ids = defaultdict(list)
-    a = mp.Aligner(reference, best_n=1, preset=preset)  # load or build index
-    if not a:
-        raise Exception("ERROR: failed to load/build index")
+
+    in_file = open(ref_sam, 'r')
+    in_sam = Reader(in_file)
 
     read_count = 0
-    for name, seq, qual in mp.fastx_read(query):  # read a fasta/q sequence
+    y = None
+    for x in in_sam:
+        if x.flag not in [0,16]:
+            continue
         read_count += 1
-        for hit in a.map(seq):  # traverse alignments
-            counts[hit.ctg] += 1
-            ranges[hit.ctg].append([hit.r_st, hit.r_en])
-            read_ids[hit.ctg].append(name)
-            # print("{}\t{}\t{}\t{}".format(name, hit.ctg, hit.r_st, hit.r_en))
-            break
-        # if read_count % 1000000 == 0:
-        #    break
+        counts[x.rname] += 1
+        ranges[x.rname].append(x.coords)
+        read_ids[x.rname].append(x.qname)
+        y = x
+
     return counts, ranges, read_ids
 
 
 def check_pileup(ref, ref_ranges, reference_file, min_coverage=0):
     if len(ref_ranges) == 0:
-        return 0, []
-    for name, seq, qual in mp.fastx_read(reference_file):
+        return 0,[]
+    for name, seq in pyfastx.Fasta(reference_file, build_index=False):
```

Comment:
What's the reasoning for iterating through the references? Isn't this info basically derivable from the sam on its own? is this sanity checking in some way?

Reply:
Mostly sanity checking, and the ref file is small so it's not a big cost. I'm generally suspicious of extracting the reference length from a sam record using a python interface in case they handle soft/hard clipped or partial matches in an unexpected way. We need to know the reference length of each reference once (which you get 0+ times from the sam file depending how many matches)

Reply:
Yeah okay that makes sense, I hadn't considered chroms with no reads